packages/permissions/src/components/authorized/authorized.js

```diff
@@ -1,66 +1,59 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { wrapDisplayName } from 'recompose';
-import { defaultMemoize } from 'reselect';
-import warning from 'warning';
-import camelCase from 'lodash.camelcase';
-import upperFirst from 'lodash.upperfirst';
 import { ApplicationContext } from '@commercetools-frontend/application-shell-connectors';
 import {
   hasSomePermissions,
   hasEveryPermissions,
+  getUnconfiguredPermissions,
 } from '../../utils/has-permissions';
-import { permissions } from '../../constants';
-
-const ensurePermissionsKeyShape = defaultMemoize(demandedPermissions =>
-  demandedPermissions.map(permission =>
-    typeof permission === 'string'
-      ? permission
-      : upperFirst(camelCase(`${permission.mode} ${permission.resource}`))
-  )
-);
 
 class Authorized extends React.Component {
   static displayName = 'Authorized';
   static propTypes = {
     shouldMatchSomePermissions: PropTypes.bool,
-    demandedPermissions: PropTypes.arrayOf(
-      PropTypes.oneOfType([
-        PropTypes.oneOf(Object.values(permissions)),
-        PropTypes.shape({
-          mode: PropTypes.string.isRequired,
-          resource: PropTypes.string.isRequired,
-        }),
-      ]).isRequired
-    ).isRequired,
+    /**
+     * This custom prop-types verifies that any permission passed as
+     * `demandedPermissions` actually exists on the `actualPermissions`.
+     *
+     * This was previously achieved through a validation via the constants (`Object.keys(permissions)`).
+     * This was deemded to not be flexible enough to introduce new permissions as they have to be
+     * added to the constants and released each time.
+     */
+    demandedPermissions(props, propName, componentName) {
+      const namesOfNonConfiguredPermissions = getUnconfiguredPermissions(
+        props.demandedPermissions,
+        props.actualPermissions
+      );
+
+      if (namesOfNonConfiguredPermissions.length > 0) {
+        return new Error(
+          `Invalid prop \`${propName}\` supplied to \`${componentName}\`. The permission(s) ${namesOfNonConfiguredPermissions.join(
```

Comment:
Nitpick: `namesOfNonConfiguredPermissions.toString()`

Reply:
Gives, `permissionA,permissionB` over `permissionA permissionB`. I don't mind I changed it to `toString`.